libraries/AP_HAL/Util.cpp

```diff
@@ -72,6 +72,23 @@ uint64_t AP_HAL::Util::get_system_clock_ms() const
 #endif
 }
 
+uint64_t AP_HAL::Util::get_system_clock_us() const
+{
+#if defined(__APPLE__) && defined(__MACH__)
+    struct timeval ts;
+    gettimeofday(&ts, nullptr);
+    return ((long long)((ts.tv_sec * 1000000) + ts.tv_usec));
+#elif CONFIG_HAL_BOARD == HAL_BOARD_CHIBIOS
+    return ST2US(chVTGetSystemTime());
```

Comment:
No scaling? This is the same as _ms() above.

Reply:
ST2MS is for ms, ST2US for us.

Reply:
aye